sei-db/sc/memiavl/db.go

```diff
@@ -204,17 +220,23 @@ func OpenDB(logger logger.Logger, targetVersion int64, opts Options) (database *
 	// create worker pool. recv tasks to write snapshot
 	workerPool := pond.New(opts.SnapshotWriterLimit, opts.SnapshotWriterLimit*10)
 
+	// Initialize lastSnapshotTime from the current snapshot directory's modification time
+	// This ensures accurate time tracking even after restarts
+	lastSnapshotTime := getSnapshotModTime(opts.Dir, mtree.Version())
```

Comment:
This is latest version which keeps changing, but we should just look at the "current" folder